subsys/net/ip/Kconfig.ipv4

```diff
@@ -34,6 +34,11 @@ config NET_DHCPV4
 	depends on NET_IPV4
 	default n
 
+config NET_IPV4_AUTOCONF
+    bool "Enable IPV4 autoconfiguration"
+    depends on NET_IPV4 && NET_ARP
+    default n
```

Comment:
You could mention the RFC 3927 here in the help text, and possible restrictions, or something that is not implemented from that rfc, etc.